webrender/src/texture_cache.rs

```diff
@@ -781,8 +781,25 @@ impl TextureCache {
         match result.kind {
             AllocationKind::TexturePage => {
                 match data {
-                    ImageData::ExternalHandle(..) => {
-                        panic!("External handle should not go through texture_cache.");
+                    ImageData::External(ext_image) => {
+                        match ext_image.image_type {
+                            ExternalImageType::Texture2DHandle |
+                            ExternalImageType::TextureRectHandle => {
+                                panic!("External texture handle should not go through texture_cache.");
+                            }
+                            ExternalImageType::ExternalBuffer => {
+                                let update_op = TextureUpdate {
```

Comment:
This block comes from:
https://github.com/servo/webrender/pull/997/commits/b4e91e1b7832ef9a983a5243042d8dd0dff9c91c#diff-d52ac938b0c681e0d7bb3f946e47489fL806